src/richie/apps/courses/api.py

```diff
@@ -45,3 +54,147 @@ def get_permissions(self):
                 permission_classes = self.permission_classes
 
         return [permission() for permission in permission_classes]
+
+
+# pylint: disable=too-many-return-statements,unused-argument, too-many-locals,too-many-branches
```

Comment:
:man_shrugging: 
Is pylint right and I should break this view? Or am I right thinking this would only make it more difficult to understand. The complexity is contained in this function and heavily tested.

Reply:
May be you can move authorization checks elsewhere but for the rest, I appreciated to not have to jump between xxx small functions to stay focus on the logic.

Reply:
:+1: 
Ok let's leave it like this for the moment and we'll factorize the authentication if we have more endpoints requiring it.